test/e2e/azure_logcollector.go

```diff
@@ -183,10 +183,21 @@ func getAzureCluster(ctx context.Context, managementClusterClient client.Client,
 	}
 
 	azCluster := &v1beta1.AzureCluster{}
-	err := managementClusterClient.Get(context.TODO(), key, azCluster)
+	err := managementClusterClient.Get(ctx, key, azCluster)
```

Comment:
This isn't strictly related to the rest of these changes, but seems like a mistake given the `ctx` parameter is otherwise not used and the other similar functions here follow this pattern.